ILSpy.BamlDecompiler/Handlers/Records/PropertyWithExtensionHandler.cs

```diff
@@ -47,7 +47,12 @@ public BamlElement Translate(XamlContext ctx, BamlNode node, BamlElement parent)
 			if (valTypeExt || extTypeId == (short)KnownTypes.TypeExtension)
```

Comment:
Wouldn't it be better to split the condition into two separate sections instead of checking `valTypeExt` twice?

Reply:
but i think that will cause more duplicated code

Reply:
do you mean this code?
``` cs
			var ext = new XamlExtension(extType);
			if (extTypeId == (short)KnownTypes.TypeExtension)
			{
				var value = ctx.ResolveType(record.ValueId);

				ext.Initializer = new object[] { ctx.ToString(parent.Xaml, value) };
			}
			else if (valTypeExt)
			{
				var value = ctx.ResolveType(record.ValueId);

				ext.Initializer = new object[] { new XamlExtension(ctx.ResolveType(0xfd4d)) { Initializer = new object[] { ctx.ToString(parent.Xaml, value) } } }; // Known type - TypeExtension
			}
			else if (extTypeId == (short)KnownTypes.TemplateBindingExtension)
			{
				var value = ctx.ResolveProperty(record.ValueId);

				value.DeclaringType.ResolveNamespace(parent.Xaml, ctx);
				var xName = value.ToXName(ctx, parent.Xaml, false);

				ext.Initializer = new object[] { ctx.ToString(parent.Xaml, xName) };
			}
			else if (extTypeId == (short)KnownTypes.StaticExtension)
			{
				string attrName;
				if (record.ValueId > 0x7fff)
				{
					bool isKey = true;
					short bamlId = unchecked((short)-record.ValueId);
					if (bamlId > 232 && bamlId < 464)
					{
						bamlId -= 232;
						isKey = false;
					}
					else if (bamlId > 464 && bamlId < 467)
					{
						bamlId -= 231;
					}
					else if (bamlId > 467 && bamlId < 470)
					{
						bamlId -= 234;
						isKey = false;
					}
					var res = ctx.Baml.KnownThings.Resources(bamlId);
					string name;
					if (isKey)
						name = res.Item1 + "." + res.Item2;
					else
						name = res.Item1 + "." + res.Item3;
					var xmlns = ctx.GetXmlNamespace(XamlContext.KnownNamespace_Presentation);
					attrName = ctx.ToString(parent.Xaml, xmlns.GetName(name));
				}
				else
				{
					var value = ctx.ResolveProperty(record.ValueId);

					value.DeclaringType.ResolveNamespace(parent.Xaml, ctx);
					var xName = value.ToXName(ctx, parent.Xaml);

					attrName = ctx.ToString(parent.Xaml, xName);
				}

				ext.Initializer = new object[] { attrName };
			}
			else if (valStaticExt)
			{
				string attrName;
				if (record.ValueId > 0x7fff)
				{
					bool isKey = true;
					short bamlId = unchecked((short)-record.ValueId);
					if (bamlId > 232 && bamlId < 464)
					{
						bamlId -= 232;
						isKey = false;
					}
					else if (bamlId > 464 && bamlId < 467)
					{
						bamlId -= 231;
					}
					else if (bamlId > 467 && bamlId < 470)
					{
						bamlId -= 234;
						isKey = false;
					}
					var res = ctx.Baml.KnownThings.Resources(bamlId);
					string name;
					if (isKey)
						name = res.Item1 + "." + res.Item2;
					else
						name = res.Item1 + "." + res.Item3;
					var xmlns = ctx.GetXmlNamespace(XamlContext.KnownNamespace_Presentation);
					attrName = ctx.ToString(parent.Xaml, xmlns.GetName(name));
				}
				else
				{
					var value = ctx.ResolveProperty(record.ValueId);

					value.DeclaringType.ResolveNamespace(parent.Xaml, ctx);
					var xName = value.ToXName(ctx, parent.Xaml);

					attrName = ctx.ToString(parent.Xaml, xName);
				}

				ext.Initializer = new object[] { new XamlExtension(ctx.ResolveType(0xfda6)) { Initializer = new object[] { attrName } } }; // Known type - StaticExtension
			}
			else
			{
				ext.Initializer = new object[] { XamlUtils.Escape(ctx.ResolveString(record.ValueId)) };
			}
```

Reply:
Oh, sorry, I missed one line.

Reply:
Fixed it...

well, I guess, we can keep it like that then.